### ISSUE TITLE
Allow clippy::drop_ref

### ISSUE THREAD

**Issue by Koxiaet** [joshua-maros/ouroboros#3]
The following code:

```rust
#[self_referencing]
struct BoxAndRef {
    data: Box<i32>,
    #[borrows(data)]
    data_ref: &'this i32,
}
```

Produces an error when run through Clippy:

```
error: calls to `std::mem::drop` with a reference instead of an owned value. Dropping a reference does nothing.
 --> src/main.rs:5:1
  |
5 | #[self_referencing]
  | ^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[deny(clippy::drop_ref)]` on by default
note: argument has type `&i32`
 --> src/main.rs:5:1
  |
5 | #[self_referencing]
  | ^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#drop_ref
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by joshua-maros** [comment]
This is now fixed in 897245a and available on crates.io as version `0.2.1`.

**Comment by joshua-maros** [comment]
*`0.2.2`, I had to correct a formatting mistake.
